.github/workflows/dependency-update.yml

```
@@ -0,0 +1,64 @@
+        
+        # Update dependencies
+        pip-compile --upgrade requirements.in
+        
+        # Check if there are changes
+        if git diff --quiet requirements.txt; then
+          echo "No dependency updates available"
+          exit 0
+        fi
+
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v5
```

actionlint (1.7.7)

the runner of "peter-evans/create-pull-request@v5" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)